model-optimizer/extensions/back/ReverseInputChannels.py

```diff
@@ -66,10 +80,25 @@ def find_and_replace_pattern(self, graph: Graph):
                       extra={'is_warning': True})
 
         for name, parameter, _ in suitable_params:
-            reverse_channels = ReverseChannels(graph, {'name': name + '/reverse_input_channels'}).create_node()
-            parameter.out_port(0).get_connection().set_source(reverse_channels.out_port(0),
-                                                              attributes_save_mode='source')
-            parameter.out_port(0).connect(reverse_channels.in_port(0))
+            reverse_index = self.get_index_for_api_2_0(parameter, 1)
+
+            # This code is needed to enable channel reversing optimization which fuses reversing to Convolution weights
+            # for case when we insert NHWC->NCHW transpose after Parameter node. Fusion does not propagate
+            # through transposes, so we insert ReverseChannels operation after the Transpose op.
+            if reverse_index != 1 and not parameter.out_port(0).disconnected():
+                out_node = parameter.out_port(0).get_connection().get_destination().node
+                if out_node.soft_get('type') == 'Transpose':
+                    order = out_node.in_port(1).data.get_value()
+                    if order is not None and order[1] == reverse_index:
+                        # At this point layout corresponds to old API (NCHW), so the axis to reverse = 1
+                        reverse_channels = ReverseChannels(graph, {'name': name + '/reverse_input_channels',
+                                                                   'axis': 1}).create_node()
+                        out_node.out_port(0).get_connection().insert_node(reverse_channels)
+                        continue
+
+            reverse_channels = ReverseChannels(graph, {'name': name + '/reverse_input_channels',
+                                                       'axis': reverse_index}).create_node()
+            parameter.out_port(0).get_connection().insert_node(reverse_channels, attributes_save_mode='source')
```

Comment:
Should we still insert ReverseChannels if parameter.out_port(0) disconnected? Possibly we can check just at the begin and ignore such parameter.

Reply:
I agree. Added check and continue.